### PR TITLE
[Android] Leverage new commit methods in Nougat

### PR DESCRIPTION
**Pull request by adrianknight89**
### Description of Change

API 24 has introduced new commit methods ([commitNow()](https://developer.android.com/reference/android/support/v4/app/FragmentTransaction.html#commitNow%28%29) and [commitNowAllowingStateLoss()](https://developer.android.com/reference/android/support/v4/app/FragmentTransaction.html#commitNowAllowingStateLoss%28%29)). I know that Xamarin Android Support Library v24 is in RC now, so this will have to wait until it's GA'ed, but I wanted to submit the work before I forget about it. In many ways, these methods should be preferable to using `executePendingTransactions()`. Nowhere in the code is `addToBackStack()` called, so these methods should not throw an `IllegalStateException`.

See [The many flavors of commit()](https://medium.com/@bherbst/the-many-flavors-of-commit-186608a015b1#.axs7wlxnd).

I haven't had a chance to test run the changes because I couldn't compile the Android Platform project with v24. I'll test it all when I can, but feel free to submit feedback now if you like.

I'll have to update this if #444 is merged.
### PR Checklist
- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [ ] Consolidate commits as makes sense

### CODE REVIEW

File: Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs [hunk 1]
```diff
@@ -612,8 +622,8 @@ Task<bool> SwitchContentAsync(Page view, bool animated, bool removed = false, bo
 					
 					Fragment toShow = fragments.Last();
 					// Execute pending transactions so that we can be sure the fragment list is accurate.
-					fm.ExecutePendingTransactions();
-					if (fm.Fragments.Contains(toShow))
+					new Handler(Looper.MainLooper).PostAtFrontOfQueue(() => FragmentManager.ExecutePendingTransactions());
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs]: Make sure this method is executed on the main thread.

File: Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs [hunk 2]
```diff
@@ -627,7 +637,14 @@ Task<bool> SwitchContentAsync(Page view, bool animated, bool removed = false, bo
...
 				}
 			}
-			transaction.Commit();
+
+			if (Build.VERSION.SdkInt >= BuildVersionCodes.N)
+				transaction.CommitNow();
+			else
+			{
+				transaction.Commit();
+				new Handler(Looper.MainLooper).PostAtFrontOfQueue(() => FragmentManager.ExecutePendingTransactions());
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs]: At the moment, I'm not sure if there are any ramifications of executing pending transactions here. I'd like to think the commits in this renderer should not be eventually executed. Similarly, `CommitNow()` should do the same job.

File: Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs [hunk 3]
```diff
@@ -124,12 +124,23 @@ protected override void Dispose(bool disposing)
...
-						FragmentManager fm = FragmentManager;
-						FragmentTransaction trans = fm.BeginTransaction();
-						foreach (Fragment fragment in _fragmentStack)
+						FragmentTransaction trans = FragmentManager.BeginTransaction();
+						trans.DisallowAddToBackStack();
+
+						while (_fragmentStack.Count != 0)
+						{
+							Fragment fragment = _fragmentStack.Last();
+							_fragmentStack.Remove(fragment);
```
Comment (adrianknight89) [inline Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs]: Removing `List` references to fragments. Not sure if this has any affect, but it somewhat bothered me that `Dispose` didn't let go of the list elements.

**Comment by jassmith** [comment]
We will come back to this when this API is ready.

**Comment by adrianknight89** [comment]
@jassmith @hartez Do you expect to update Android support libraries anytime soon so the suggestions here can be incorporated?

Also see https://forums.xamarin.com/discussion/84677/remove-specific-version-dependencies-of-android-support-libraries#latest.

**Comment by rmarinho** [comment]
@adrianknight89 we have a PR in testing for that check #1107

**Comment by rmarinho** [comment]
ohh and @adrianknight89 your suggestion is already on the product, we don't block the platform support library https://github.com/xamarin/Xamarin.Forms/blob/master/.nuspec/Xamarin.Forms.nuspec#L27

**Comment by rmarinho** [comment]
Btw can we rebase this if still makes sense ? :) thanks

**Comment by adrianknight89** [comment]
@rmarinho Thanks. I'll go over this again when #1107 is merged. I need to make sure everything works as intended. :)

**Comment by rmarinho** [comment]
Hi @adrianknight89 now that #1107 is merged can you take a look at this and rebase? 

Thanks

**Pull request closed without merging**